src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -449,6 +460,168 @@ export default class SourceFileLinter {
 		}
 	}
 
+	analyzeComponentJson(node: ts.ExpressionWithTypeArguments) {
+		if (node.expression.getText() !== "UIComponent") {
```

Comment:
I'm not sure whether we can handle such situations.
The reasoning is that we do not have a mechanism to analyse the content of the dependencies so far. 
Further more, the dependency could be an external lib/app and in such cases we cannot even access the code/type definitions for it.

I saw the `sap/fe/core/AppComponent`. But as long as it has this interface everything is clear there and we shouldn't report anything. Its children would inherit that and our check would not analyze them as they do not inherit from `UIComponent`
So far the hardcoded option could be the only solution and the benefit from it would be to warn developers when they have `async` flags in their manifests, so they can clean up.
The real issue would have been if the `sap/fe/core/AppComponent` didn't have that interface, but other components were inheriting from it. Then, maybe the hardcoded solution would be the only option so far.

Any feedback on how to workaround this limitation would be highly appreciated!

Reply:
> the AppComponent class is special in a way as it implements the "IAsyncContentCreation" interface, but that interface is not reflected in sap.fe.core.d.ts (to me it is currently unknown why this is the case). 

I suppose that it's because of the UI5 version. Maybe, they have impolemented that interface after 1.120.x

Reply:
> I suppose that it's because of the UI5 version. Maybe, they have impolemented that interface after 1.120.x

The interface is defined there already for a while, so I suspect it's an issue with generating the d.ts.

> The real issue would have been if the sap/fe/core/AppComponent didn't have that interface, but other components were inheriting from it. Then, maybe the hardcoded solution would be the only option so far.

Then let's use [APF](https://ui5.sap.com/#/topic/0109e67712bc4966bfa8922909e92a87) as another example of a base component class. It does not define the interface and ideally we should detect component best practices for that use case.
I hope you get my point. We should check the inheritance chain to see whether a `Component.js` module inherits from `UIComponent`.

Reply:
Now the soluton uses type definitions and is able to go down to the Base class. As we need to detect only `UIComponent`, the soltion is limited up to it.